frontend/src/libs/hooks/use-homepage-socket-service/use-homepage-socket-service.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  ClientSocketEvent,
+  ServerSocketEvent,
+  socket as socketService,
+} from '~/libs/packages/socket/socket.js';
+import { type TruckEntity } from '~/packages/trucks/libs/types/types.js';
+import { actions } from '~/slices/trucks/trucks.js';
+
+import { useAppDispatch, useCallback } from '../hooks.js';
+
+type ReturnType = {
+  connectToHomeRoom: () => void;
+  disconnectFromHomeRoom: () => void;
+};
+
+const useHomePageSocketService = (): ReturnType => {
+  const dispatch = useAppDispatch();
+
+  const connectToHomeRoom = useCallback((): void => {
+    socketService.emit({
+      event: ClientSocketEvent.JOIN_HOME_ROOM,
+    });
+    socketService.addListener(
+      ServerSocketEvent.TRUCKS_LIST_UPDATE,
+      (payload): void => {
+        void dispatch(actions.setTrucks(payload as TruckEntity[]));
+      },
+    );
+  }, [dispatch]);
+
+  const disconnectFromHomeRoom = useCallback((): void => {
+    socketService.emit({
+      event: ClientSocketEvent.LEAVE_HOME_ROOM,
+    });
+  }, []);
+
```

Comment:
logic of emiting and listening socket events can be moved to redux middelware, then listeners will not be created on components updates

Reply:
I'll change that when your PR gets accepted, since you've already implemented the middleware.